Makefile-v2.mk

```diff
@@ -0,0 +1,296 @@
+## Build, test, and generate code for various parts of Grafana Agent.
+##
+## At least Go 1.18, git, and a moderately recent version of Docker is required
```

Comment:
might want to mention protoc is required as well to run generate-protos

Reply:
Yeah, this wasn't supposed to be an exhaustive list since it'll probably get out of sync pretty fast. I'll elaborate the comment, thanks!